pkg/telemetry/telemetry.go

```diff
@@ -268,6 +275,60 @@ func (t *engineImpl) syncInfoTable(stats map[string]float64) error {
 	return nil
 }
 
+func (t *engineImpl) housekeeping() {
+	var lastRunBeyondHour bool
+	err := t.conn.QueryRow(context.Background(), "SELECT _ps_catalog.promscale_telemetry_housekeeping()").Scan(&lastRunBeyondHour)
+	if err != nil && !strings.Contains(err.Error(), "could not obtain lock on relation") { // Ignore lock errors as we aren't supposed to do anything, since some other Promscale is doing housekeeping at this time.
```

Comment:
Is this error text the same across major versions? If not, it might be preferable to catch the exception in the `promscale_telemetry_housekeeping` function and return an additional boolean that reflects whether or not the attempt was successful

Reply:
It might be cleaner to use the error code `55P03` rather than the error message.
`ERROR:  55P03: could not obtain lock on relation "foo"`

https://www.postgresql.org/docs/14/errcodes-appendix.html

Reply:
https://github.com/jackc/pgconn/blob/v1.10.1/errors.go#L33

Reply:
Can we catch this error in the SQL function + return false?

Reply:
I wrote this after the lock statement, but it does not seem to work.

```postgresql
EXCEPTION
            WHEN SQLSTATE '55P03' THEN -- sqlstate when lock on table is not available.
                RETURN FALSE;
```